csharp/src/Apache.Arrow/Schema.cs

```diff
@@ -63,5 +63,48 @@ public int GetFieldIndex(string name, StringComparer comparer = default)
             return _fields.IndexOf(
                 _fields.Single(x => comparer.Compare(x.Name, name) == 0));
         }
+
+        public override bool Equals(object obj)
+        {
+            if (this == obj)
+            {
+                return true;
+            }
+
+            if (obj == null || !this.GetType().Equals(obj.GetType()))
+            {
+                return false;
+            }
+
+
+            Schema other = (Schema)obj;
+            var otherFields = other.Fields;
+
+            if (this.HasMetadata != other.HasMetadata)
+            {
+                return false;
+            }
+            if (this.Fields.Count != otherFields.Count)
+            {
+                return false;
+            }
+            foreach (var pair in Fields)
```

Comment:
I wonder if you should check the `_fields` property instead. It is a List. So you can compare that the Schemas have the same order of Fields as well.